src/components/views/rooms/wysiwyg_composer/components/WysiwygComposer.tsx

```diff
@@ -47,21 +53,53 @@ export const WysiwygComposer = memo(function WysiwygComposer({
     rightComponent,
     children,
 }: WysiwygComposerProps) {
-    const inputEventProcessor = useInputEventProcessor(onSend, initialContent);
+    const { room } = useRoomContext();
+    const autocompleteRef = useRef<Autocomplete | null>(null);
 
-    const { ref, isWysiwygReady, content, actionStates, wysiwyg } = useWysiwyg({ initialContent, inputEventProcessor });
+    const inputEventProcessor = useInputEventProcessor(onSend, autocompleteRef, initialContent);
+    const { ref, isWysiwygReady, content, actionStates, wysiwyg, suggestion } = useWysiwyg({
+        initialContent,
+        inputEventProcessor,
+    });
+    const { isFocused, onFocus } = useIsFocused();
+
+    const isReady = isWysiwygReady && !disabled;
+    const computedPlaceholder = (!content && placeholder) || undefined;
+
+    useSetCursorPosition(!isReady, ref);
 
     useEffect(() => {
         if (!disabled && content !== null) {
             onChange?.(content);
         }
     }, [onChange, content, disabled]);
 
-    const isReady = isWysiwygReady && !disabled;
-    useSetCursorPosition(!isReady, ref);
+    useEffect(() => {
+        function handleClick(e: Event): void {
+            e.preventDefault();
+            if (
+                e.target &&
+                e.target instanceof HTMLAnchorElement &&
+                e.target.getAttribute("data-mention-type") === "user"
+            ) {
+                const parsedLink = parsePermalink(e.target.href);
+                if (room && parsedLink?.userId)
+                    defaultDispatcher.dispatch({
+                        action: Action.ViewUser,
+                        member: room.getMember(parsedLink.userId),
+                    });
+            }
+        }
 
-    const { isFocused, onFocus } = useIsFocused();
-    const computedPlaceholder = (!content && placeholder) || undefined;
+        const mentions = ref.current?.querySelectorAll("a[data-mention-type]");
+        if (mentions) {
+            mentions.forEach((mention) => mention.addEventListener("click", handleClick));
+        }
+
+        return () => {
+            if (mentions) mentions.forEach((mention) => mention.removeEventListener("click", handleClick));
+        };
+    }, [ref, room, content]);
```

Comment:
We get link tags (representing mentions) from the rust model in a string of HTML that gets set as inner HTML on any update to the rust model.

We will be able to style the links using attributes they have, but for click handling we need to add/remove the handlers on any change of the rust model output.